### DISCUSSION TITLE
Validating the caching concept

### DISCUSSION THREAD

**Discussion by dev-tarun-nw** [vercel/turborepo#9097] [Help] [answered]
### Summary

Hi,
I have setup a turbo-repo for my project, in their is no separate build process define for packages these packages will building along with the app.

app
|_ app1
packages
|_ui

Now, i have few doubt related to this : 
CASE: Only App1 files is changed
* This app1 have dependency with package ui, will cause ui package files to rebuild again or only app1 rebuild ?

**Comment by anthonyshew** [comment, marked as answer]
Task ordering and caching behavior will follow your Task and Package Graphs. More information here: https://turbo.build/repo/docs/core-concepts/package-and-task-graph
